charmcraft/charm_builder.py

```diff
@@ -224,21 +229,23 @@ def handle_dependencies(self):
 
         # virtualenv with other dependencies (if any)
         if self.requirement_paths:
-            _process_run(["pip3", "--version"])
-
-            venvpath = self.buildpath / VENV_DIRNAME
-            cmd = [
-                "pip3",
-                "install",  # base command
-                "--target={}".format(venvpath),  # put all the resulting files in that specific dir
-            ]
+            staging_venv_dir = self.charmdir / STAGING_VENV_DIRNAME
+            _process_run(["python3", "-m", "venv", str(staging_venv_dir)])
+            pip_cmd = str(staging_venv_dir / "bin" / "pip3")
```

Comment:
Using a modified version of `find_venv_dir` that does the appropriate processing based on platform instead of verifying if the directory exists will probably suit us better and also be more test-friendly since the venv creation process is mocked up. I'll prepare a patch.